cypress/integration/pages/onDemandAudio/tests.js

```diff
@@ -83,40 +84,79 @@ export default ({ service, pageType, variant, isAmp }) => {
                   const episodeId = path(['content', 'blocks', 0, 'id'], body);
 
-                  const expectedNumberOfEpisodes = processRecentEpisodes(body, {
+                  const processedEpisodesData = processRecentEpisodes(body, {
                     exclude: episodeId,
                     recentEpisodesLimit: recentEpisodesMaxNumber,
-                  }).length;
+                  });
+
+                  const expectedNumberOfEpisodes = processedEpisodesData.length;
 
                   cy.log(
                     `Number of available episodes? ${expectedNumberOfEpisodes}`,
                   );
-                  // More than one episode expected
-                  if (expectedNumberOfEpisodes > 1) {
-                    cy.get('[data-e2e=recent-episodes-list]').should('exist');
-
-                    cy.get('[data-e2e=recent-episodes-list]').within(() => {
-                      cy.get('[data-e2e=recent-episodes-list-item]')
-                        .its('length')
-                        .should('eq', expectedNumberOfEpisodes);
-                    });
-                  }
-                  // If there is only one item, it is not in a list
-                  else if (expectedNumberOfEpisodes === 1) {
-                    cy.get('aside[aria-labelledby=recent-episodes]').within(
-                      () => {
-                        cy.get('div[class*="Wrapper"]').should('exist');
-                      },
+
+                  cy.window().then(win => {
+                    const renderedEpisodes = win.document.querySelectorAll(
+                      '[data-e2e=recent-episodes-list-item]',
                     );
-                  }
-                  // No items expected
-                  else {
-                    cy.get('aside[aria-labelledby=recent-episodes]').should(
-                      'not.exist',
+
+                    const renderedEpisodesArray = Array.prototype.slice.call(
+                      renderedEpisodes,
+                    );
+
+                    const renderedEpisodesInnerText = renderedEpisodesArray.map(
+                      episode => episode.innerText,
                     );
 
-                    cy.log('No episodes present or available');
-                  }
+                    const jsonResWithHumanReadableDates = processedEpisodesData.map(
+                      episodeData => ({
+                        ...episodeData,
+                        timestamp: new Date(
+                          episodeData.timestamp,
+                        ).toLocaleString(),
+                      }),
+                    );
+
+                    if (
+                      renderedEpisodesArray.length !==
+                      jsonResWithHumanReadableDates.length
+                    ) {
+                      /* eslint-disable no-console */
+                      console.log(
+                        'From json response - ',
+                        jsonResWithHumanReadableDates,
```

Comment:
Should we log the equivalent value from `win.SIMORGH_DATA.data` too?

Would allow us to confirm (or reject) the caching hypothesis if the JSON response received by Cypress and the JSON response received by Simorgh are different (or are the same)

Reply:
That's a great idea man. We talked ourselves out of using window.SIMORGH_DATA after you left but this makes a lot of sense.

Reply:
👍  would just need some null protection to account for `SIMORGH_DATA` being undefined on amp

Reply:
Yo added it Rydawg wanna take a 👁️